src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -1026,6 +1027,56 @@ def list_roles(self) -> List[Dict[str, Any]]:
 
         return roles
 
+    def create_role(self, role_name: str, role_path: str, role_assume_policy_doc: str) -> None:
+        try:
+            json.loads(role_assume_policy_doc)
+        except:  # noqa: E722
+            raise DashboardException('Assume role policy document is not a valid json')
+
+        def between(v, a, b):
+            return ord(v) >= ord(a) and ord(v) <= ord(b)
+        # valid values:
+        # pylint: disable=C0301
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-path # noqa: E501
+        valid_special = '_+=,.@'
+        if len(role_name) > 64:
+            raise DashboardException(
+                f'Role name "{role_name}" is invalid. Should be 64 characters or less')
+        for c in role_name:
+            if not (between(c, 'a', 'z')
+                    or between(c, 'A', 'Z')
+                    or between(c, '0', '9')
+                    or c in valid_special):
+                raise DashboardException(
+                    f'Role name "{role_name}" is invalid. Valid characters are [a-zA-Z_+=,.@]')
```

Comment:
Same thing, why not testing against that regex? Regex are [systematically a bad idea except for when they are not](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/).